### ISSUE TITLE
[HOLD for payment 2024-07-10] [HOLD for payment 2024-06-20] [HOLD for payment 2024-06-18] [$250] Pay someone flow doesn’t include a reportAction to distinguish the payment method used (Expensify or elsewhere)

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#41645]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.70-2
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @trjExpensify 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1714729020479179

## Action Performed:
Prerequisite: Pay someone with either Expensify or elsewhere option
1. Login as a user who got paid
2. Open the DM and observe the preview for money received
3. Click to open the payment details and observe
## Expected Result:
Method of payment displayed in both DM and details page (Expensify or elsewhere)
## Actual Result:
Only xyz paid displayed and method of payment not displayed
## Workaround:
Unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [X] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos
![image (1)](https://github.com/Expensify/App/assets/38435837/c7f65b6a-44c5-4163-93bb-5b1686dfed73)
![Snip - (2) New Expensify](https://github.com/Expensify/App/assets/38435837/d00143cf-b4a8-41de-990b-82ba27608d88)

Add any screenshot/video evidence

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~014f39f2dacd55f2a4</li>
        <li>Upwork Job ID: 1787959016608243712</li>
        <li>Last Price Increase: 2024-05-21</li>
<li>Automatic offers: </li>
<ul>
<li>ahmedGaber93 | Reviewer | 102528931</li>
<li>nkdengineer | Contributor | 102528935</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @johncschuster</details>

**Comment by ahmedGaber93** [comment]
@jainilparikh Thanks for the proposal.

I don't think we need to add a new field for payment type, we need to edit the header and IOU preview to be `xyz paid $50.00 using Expensify` instead of `xyz paid $50.00`

![Screenshot 2024-05-09 at 3 52 39 PM](https://github.com/Expensify/App/assets/41129870/5046cc47-a743-41a6-9e4c-61ea3222c56a)
![Screenshot 2024-05-09 at 3 52 22 PM](https://github.com/Expensify/App/assets/41129870/89c40974-5a0d-4e2a-bf96-c13b6e8cf22c)

**Comment by sakluger** [comment]
Not overdue, we're still discussing the expected behavior.

**Comment by ahmedGaber93** [comment]
I checked it today, we already have the paymentType in FE but not with this reportAction, I will complete checking tomorrow to know if we can catch it from FE, or it requires BE changes

**Comment by sakluger** [comment]
Reassigning while I'm OOO for two weeks.

**Comment by sakluger** [comment]
Al will also be OOO for the same time, re-reassigning!

**Comment by ahmedGaber93** [comment]
I don't think we need to BE changes as we already have `paymentType` in FE, and we can use it.

**Comment by ahmedGaber93** [comment]
> @ahmedGaber93 , the reportAction does not have any information about how payment is done (Via Expensify or via Elsewhere).
Do we first need the BE to send this info in the parentReportActions ?

@jainilparikh we already have it in another place, so no need to BE changes.

**Comment by ahmedGaber93** [comment]
@trjExpensify can you please confirm is this the expected behavior?

![Screenshot 2024-05-17 at 1 29 15 AM](https://github.com/Expensify/App/assets/41129870/fb443edc-c989-4014-87bb-fe21c16a97b1)




 or this https://github.com/Expensify/App/issues/41645#issuecomment-2102612537?

**Comment by trjExpensify** [comment]
What you're showing in that screenshot is what's expected. The report is missing the payment reportAction.

**Comment by ahmedGaber93** [comment]
I can't reproduce in latest main. Are I missed something? @trjExpensify 


https://github.com/Expensify/App/assets/41129870/34036123-024d-44f4-91a8-1fe55500ddfd

**Comment by trjExpensify** [comment]
Yeah, you need to use the `Pay someone` flow in global create. Not the `Submit expense` flow and then pay a request.

**Comment by ahmedGaber93** [comment]
I can reproduce now. Thanks!


https://github.com/Expensify/App/assets/41129870/42b6f575-012d-44e5-b25e-d7ca9d13ca1d

**Comment by ahmedGaber93** [comment]
@jainilparikh your proposal still not cover the expected result, the payment type should be `reportAction` not new field in IOU details page

![331402889-fb443edc-c989-4014-87bb-fe21c16a97b1](https://github.com/Expensify/App/assets/41129870/1a40f604-137b-44dc-b9d2-5a798d380ea7)

**Comment by ahmedGaber93** [comment]
Waiting on proposals

**Comment by johncschuster** [comment]
[Posted in #expensify-open-source to drum up proposals](https://expensify.slack.com/archives/C01GTK53T8Q/p1716328788343329)

**Comment by nkdengineer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Only xyz paid displayed and method of payment not displayed

### What is the root cause of that problem?
When we send money, we don't create pay action in optimistic data of `iouReport` and BE doesn't return the pay report action when we send money request.

https://github.com/Expensify/App/blob/e8ae3c5acedf0e6788dc574c7b6f3043ca37092a/src/libs/actions/IOU.ts#L5759


### What changes do you think we should make in order to solve the problem?
1. We should create pay report action in optimistic data when we send money. The way to create this action is the same as we do when we pay a money request [here](https://github.com/Expensify/App/blob/e8ae3c5acedf0e6788dc574c7b6f3043ca37092a/src/libs/actions/IOU.ts#L5790)

https://github.com/Expensify/App/blob/e8ae3c5acedf0e6788dc574c7b6f3043ca37092a/src/libs/actions/IOU.ts#L5759

2. Accept an extra param like `paidReportActionID` that is the `reportActionID` of the report action that we created at the point 1

https://github.com/Expensify/App/blob/e8ae3c5acedf0e6788dc574c7b6f3043ca37092a/src/libs/actions/IOU.ts#L5759

3. We also need to fix in BE side to create a paid action with the `paidReportActionID` params in `SendMoneyElsewhere` and `SendMoneyWithWallet` API
### What alternative solutions did you explore? (Optional)
NA

**Comment by ahmedGaber93** [comment]
Reviewing today

**Comment by ahmedGaber93** [comment]
@nkdengineer Thanks for the proposal.

I don't think your RCA is correct, as we already create optimistic pay action `CONST.IOU.REPORT_ACTION_TYPE.PAY` here

https://github.com/Expensify/App/blob/8375abea3a377e7416404addde255b6ac8530a9b/src/libs/actions/IOU.ts#L5561-L5564

**Comment by nkdengineer** [comment]
@ahmedGaber93 Thanks for your feedback, I updated proposal https://github.com/Expensify/App/issues/41645#issuecomment-2124381544

**Comment by ahmedGaber93** [comment]
@nkdengineer Thanks for the update.

@nkdengineer's [proposal](https://github.com/Expensify/App/issues/41645#issuecomment-2124381544) using cancel filtering "send money pay action" in `combinedReportActions` then display it as a normal message not money request preview in expense report page LGTM.

🎀 👀 🎀 C+ reviewed

**Comment by ahmedGaber93** [comment]
@stitesExpensify please take a look here when you have a time.

**Comment by stitesExpensify** [comment]
> then display it as a normal message not money request preview in expense report page

Can you post screenshots of what it used to look like, and what it will look like now? This seems like a significant design change

**Comment by nkdengineer** [comment]
@stitesExpensify This will show the message like the result in my proposal here https://github.com/Expensify/App/issues/41645#issuecomment-2124381544

**Comment by stitesExpensify** [comment]
Okay, I see that now at the end. What does it look like currently?

**Comment by nkdengineer** [comment]
@stitesExpensify  Actually, Nothing is shown in combine report of send money request because the send money action is filtered out.

<img width="1507" alt="Screenshot 2024-05-30 at 01 59 02" src="https://github.com/Expensify/App/assets/161821005/b13551f6-7eb0-4553-be66-c9e7a4c5d413">

**Comment by stitesExpensify** [comment]
So it was never showing `money request preview` like [this comment ](https://github.com/Expensify/App/issues/41645#issuecomment-2132121945)suggested?

**Comment by ahmedGaber93** [comment]
> So it was never showing money request preview like https://github.com/Expensify/App/issues/41645#issuecomment-2132121945suggested?

@stitesExpensify Yes, It was never showing because we filter it [here](https://github.com/Expensify/App/blob/8375abea3a377e7416404addde255b6ac8530a9b/src/libs/ReportActionsUtils.ts#L320-L324), but now In this issue, we are going to cancel filtering it because it has the payment method and message we need `Paid USD 12.00 elswhere`. 


**The problem now is after cancel filtering it, it will display as `money request preview`** 

https://github.com/Expensify/App/blob/8375abea3a377e7416404addde255b6ac8530a9b/src/libs/ReportActionsUtils.ts#L320

so the point 2 in @nkdengineer's solution is to display it as a normal message `Paid USD 12.00 elswhere` not `money request preview`.

**Comment by stitesExpensify** [comment]
Great, thank you both for helping me catch up! Assigning @nkdengineer

**Comment by nkdengineer** [comment]
@ahmedGaber93 The PR is [here](https://github.com/Expensify/App/pull/42894).

**Comment by ahmedGaber93** [comment]
@nkdengineer the PR is reverted because of this regression https://github.com/Expensify/App/issues/43183. Can you investigate the issue?

**Comment by nkdengineer** [comment]
@ahmedGaber93 Coming from https://github.com/Expensify/App/issues/43183 I want to discuss again about the expected behavior because the child report of send money request action is transaction thread report:

1. Not display as combine report for iou report of send money request so send money request action will display as preview and click on this will open the transaction thread report

2. Still display send money request as Paid system message and fix this issue https://github.com/Expensify/App/issues/43183 to display the child report of send money action as normal thread

I prefer option 1 since the child report is transaction thread report, it doesn't make sense if we display it as a normal thread

Please help to confirm the expected behavior that we want here cc @trjExpensify @stitesExpensify

**Comment by trjExpensify** [comment]
I'm not sure I'm 100% following the question as asked, but in essence, it should be identical to the one expense report in the `submit expense` flow.

**Comment by ahmedGaber93** [comment]
@nkdengineer any updates?

**Comment by nkdengineer** [comment]
I will create another PR early tomorrow morning.

**Comment by nkdengineer** [comment]
@ahmedGaber93 The PR is [here](https://github.com/Expensify/App/pull/43742).

**Comment by johncschuster** [comment]
We're waiting on the linked PR above to be merged

**Comment by johncschuster** [comment]
Still waiting on the linked PR to cross the regression threshold 👍

**Comment by johncschuster** [comment]
Same as above

**Comment by johncschuster** [comment]
Same as above

**Comment by johncschuster** [comment]
Looks like there was a regression on this one. @stitesExpensify can you see if the linked PR caused the regression?

**Comment by stitesExpensify** [comment]
Based on the discussion in that issue, tt looks like this PR is the cause, but it might have just needed _more_ as opposed to needing to be reverted etc. @nkdengineer @ahmedGaber93 can you both make sure to keep an eye on that discussion since this did technically cause a regression?

**Comment by ahmedGaber93** [comment]
The regression case currently in discussion if it should fix or not. And I am 👀  on it.

**Comment by nkdengineer** [comment]
I checked this issue and I think it's not regression from our PR, because the transaction thread report was hidden in the past.

**Comment by johncschuster** [comment]
Acknowledged! I'll issue payment after the regression window.

**Comment by johncschuster** [comment]
@ahmedGaber93 can you please complete the BZ Checklist?

**Comment by ahmedGaber93** [comment]
**BugZero Checklist**: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [x] [@ahmedGaber93] The PR that introduced the bug has been identified. Link to the PR: N/A. Pay action in send money is not implemented before, so this seems a new requirement
- [x] [@ahmedGaber93] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: N/A.
- [x] [@ahmedGaber93] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: N/A.
- [x] [@ahmedGaber93] Determine if we should create a regression test for this bug. Yes
- [x] [@ahmedGaber93] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.



### Regression Test Proposal

1. Go to a DM and create a send money request
2. Click on report preview to open iou report
3. Verify that a paid system message appears
4. Reply in thread this system message
5. Verify that the thread is displayed as a normal thread

Do we agree 👍 or 👎

**Comment by ahmedGaber93** [comment]
@johncschuster completed

**Comment by johncschuster** [comment]
Payment has been issued! Thanks for your contributions!
